tests/artifactorycustomhttpclient_test.go

```diff
@@ -0,0 +1,37 @@
+package tests
+
+import (
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/config"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/url"
+	"testing"
+)
+
+func TestGetArtifactoryVersionWithProxyShouldFail(t *testing.T) {
```

Comment:
This test can also fail due to bad configured test Artifactory URL, and not only because of the dummy proxy.
Here's what I suggest - 
1. Let's get the test to get the version, without setting the custom proxy, and validate success.
2. Then set the dummy proxy and validate failure.

Reply:
I think you're already doing #1 in a different test file (https://github.com/jfrog/jfrog-client-go/blob/master/tests/artifactory_test.go#L6-L16) but I've added an additional test that ensures we can download files with a default client.